### DISCUSSION TITLE
How to debug?

### DISCUSSION THREAD

**Discussion by konsolebox** [util-linux/util-linux#3323] [Q&A]
Are there any specific configurations needed to enable debug lines from util-linux?  Also what message channel are they sent?  E.g., logfile, syslog, etc.  I think these things should be added in the README.

Tips on how to insert new debug lines would be helpful too.  This saves the debugger's time.

**Comment by karelzak** [comment]
Well, util-linux is a large set of tools with different use cases. I do not think there is a generic answer.

* shared libraries usually use LIBxxxx_DEBUG=all env. variable to print debug messages on stderr (e.g. `LIBMOUNT_DEBUG=all findmnt`).
* debugger stuff can be enabled by adding `-g` to CFLAGS, for example, by `CFLAGS="-g" ./configure`
* some tools provide verbose output, for example, hardlink
* some tools need code modification (e.g. agetty)

**Comment by brianjmurrell** [comment]
Maybe a more specific question: how does one get the `DBG()` code in https://github.com/util-linux/util-linux/blob/master/lib/fileeq.c#L510-L555 to execute for example?

**Reply by karelzak** [reply]
`ULFILEEQ_DEBUG=all` hardlink ...

I have added a note about this env. variable to the hardlink man page.
